packages/esm-sms-app/src/hooks/useOverlay.ts

```diff
@@ -0,0 +1,47 @@
+import { getGlobalStore, ExtensionSlot, renderExtension } from '@openmrs/esm-framework';
+import { useEffect, useState } from 'react';
+
+interface OverlayStore {
+  isOverlayOpen: boolean;
+  component?: any;
+  header: string;
+}
+
+const initialState = { isOverlayOpen: false, component: Function, header: '' };
+
+const getOverlayStore = () => {
```

Comment:
Shouldn't this be using either the Modal or Workspace system available in core? I don't know why we'd need another.

Reply:
If we stick with this, use a more specific name than `providers-store`. Global stores (as the name suggests) are global and non-unique names will cause issues down the line. It isn't obvious that something called `sms-app` would "own" the store called `providers-store`.

Reply:
> Shouldn't this be using either the Modal or Workspace system available in core? I don't know why we'd need another.

I tried using both the workspace window and overlay but they didn't fit perfectly. I think the workspace window is designed specifically for the patient chart, so made a crazy layout. Whereas the workspace overlay was, as you know, covering the contents below it. 



> If we stick with this, use a more specific name than `providers-store`. Global stores (as the name suggests) are global and non-unique names will cause issues down the line. It isn't obvious that something called `sms-app` would "own" the store called `providers-store`.

is it okay to have the app feature name as the store  name?

Reply:
The workspace isn't _just_ intended to be used in the chart. It's made to be used across the app. The overlay does cover things, I suppose. Can the component you're rendering be worked to fit in the size constraints of the workspace?

> is it okay to have the app feature name as the store name?

It's preferable even.

Reply:
> The workspace isn't _just_ intended to be used in the chart. It's made to be used across the app. The overlay does cover things, I suppose. Can the component you're rendering be worked to fit in the size constraints of the workspace?
> 
> > is it okay to have the app feature name as the store name?
> 
> It's preferable even.



https://github.com/user-attachments/assets/a735e0a2-22a5-4151-9f1b-d14b30491724

Reply:
I think am resorting to this. I have used `sms-app-store` as the store name

Reply:
@Jexsie It's not clear to me from looking at the video what the problem is, or what you mean by "made a crazy layout." Could you post screenshots or video so we can see what the problem is? This kind of thing is exactly what the workspace system is for. If we figure out what the problem you're having is maybe we can improve the documentation so others don't run into the same problem.

Reply:
The video attached above shows the expected layout  that i wanted to achieve using the custom overlay I added.   The `worskpaceOverlay` wasn't giving me that.

Reply:
@Jexsie Can you post a screen shot and (ideally) some code using the workspace system that's not achieving what you'd want?

Reply:
Sorry for the delayed response, the cfl server was down. Below is what happens if I use the `workspaceWindow`. The forms are open on the left,  plus some whitespace is added to the bottom

### Workspace WIndow

https://github.com/user-attachments/assets/038fb251-2208-407d-8f76-b093ef4701e3

### Custom overlay created

https://github.com/user-attachments/assets/a5054f33-2c29-4fc9-a678-e6ca881ac95